### PR TITLE
Implement MVVM for the Interaction page

### PR DESCRIPTION
**Pull request by PankajBhojwani**
## Summary of the Pull Request
Implements an `InteractionViewModel` for the Interaction page in the SUI

## PR Checklist
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Validation Steps Performed
Interaction page still works

### CODE REVIEW

File: src/cascadia/TerminalSettingsEditor/InteractionViewModel.idl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Licensed under the MIT license.
+
+import "EnumEntry.idl";
+
+namespace Microsoft.Terminal.Settings.Editor
+{
+    runtimeclass InteractionViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
+    {
+        InteractionViewModel(Microsoft.Terminal.Settings.Model.GlobalAppSettings globalSettings);
+        Microsoft.Terminal.Settings.Model.GlobalAppSettings GlobalSettings;
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/InteractionViewModel.idl:11]:
```suggestion
        Microsoft.Terminal.Settings.Model.GlobalAppSettings GlobalSettings{ get; };
```
Do we need to expose the setter for this?

**Review by carlos-zamora — APPROVED** [review]
Looks good to me! This page doesn't really have any tricky parts, thankfully.

File: src/cascadia/TerminalSettingsEditor/InteractionViewModel.idl (hunk 1 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/InteractionViewModel.idl:11]: Whoops, thanks for the catch!

**Review by lhecker — APPROVED** [review]

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Blocking until 1.15 forks (I approve this PR in its current state though)

**Review by zadjii-msft — APPROVED** [review]
These are all gonna be pretty straightforward, aren't they

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/Interaction.xaml [hunk 2]
```diff
@@ -28,64 +28,64 @@
         <StackPanel Style="{StaticResource SettingsStackStyle}">
             <!--  Copy On Select  -->
             <local:SettingContainer x:Uid="Globals_CopyOnSelect">
-                <ToggleSwitch IsOn="{x:Bind State.Globals.CopyOnSelect, Mode=TwoWay}"
+                <ToggleSwitch IsOn="{x:Bind ViewModel.GlobalSettings.CopyOnSelect, Mode=TwoWay}"
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Interaction.xaml:31]: Shouldn't the viewmodel abstract away the need to access the actual model?

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Interaction.xaml:31]: Fixed!

**Pull request merged**